### PR TITLE
Cart info in tallying phase incorrect

### LINKED ISSUE #104: Cart info in tallying phase incorrect
image

I think the cart remembers the items you had in your cart when the reallocation phase ended, not how you actually contributed... 

After reallocation the cart should transition to your last contributed cart.

### PR DESCRIPTION
**Pull request by corwintines**
Fixes: https://github.com/ethereum/clrfund/issues/104

Testing cases:
- Make sure tallying phase still shows correct committed items
- Remove all cart items in reallocation, make sure tallying phase still shows correct committed items
- Edit cart but dont commit while in reallocation, make sure tallying phase still shows correct committed items
- Edit cart in reallocation phase, make sure tallying phase still shows correct committed items
- Refresh app, make sure tallying phase still shows correct committed items

### CODE REVIEW

File: vue-app/src/components/Cart.vue [hunk 1]
```diff
@@ -409,6 +419,19 @@ export default class Cart extends Vue {
...
+      return this.$store.state.committedCart.reduce((total: BigNumber, item: CartItem) => {
+        let amount
+        try {
+          amount = parseFixed(item.amount, nativeTokenDecimals)
+        } catch {
+          return total
+        }
+        return total.add(amount.div(voiceCreditFactor).mul(voiceCreditFactor))
+      }, BigNumber.from(0))
+    }
```
Comment (pettinarip) [inline vue-app/src/components/Cart.vue]: Just a nice to have refactor: 

Maybe we could add a new method called `getCartTotal` to abstract the calculation and reuse it below. Like this:

```
getCartTotal(cart: Cart): BigNumber {
  const { nativeTokenDecimals, voiceCreditFactor } = this.$store.state.currentRound
  return cart.reduce((total: BigNumber, item: CartItem) => {
    let amount
    try {
      amount = parseFixed(item.amount, nativeTokenDecimals)
    } catch {
      return total
    }
    return total.add(amount.div(voiceCreditFactor).mul(voiceCreditFactor))
  }, BigNumber.from(0)) 
}
```
then, the `getTotal` would looks like this:
```
getTotal(): BigNumber {
  const { cart, committedCart, currentRound } = this.$store.state

  // In tally round, use the committedCart to get total contribution amount
  if (currentRound.status === RoundStatus.Tallying) {
    return this.getCartTotal(committedCart)
  }

  return this.getCartTotal(cart)
}
```

File: vue-app/src/components/Cart.vue [hunk 2]
```diff
@@ -346,6 +352,10 @@ export default class Cart extends Vue {
   }
 
   get filteredCart(): CartItem[] {
+    // In tallying round use committedCart for cart items
+    if (this.$store.state.currentRound.status === RoundStatus.Tallying) {
```
Comment (samajammin) [inline vue-app/src/components/Cart.vue]: I think we'll want this for any round status beyond reallocation. I believe there's a "hasReallocationElapsed" getter or similar.

File: vue-app/src/components/Cart.vue [hunk 3]
```diff
@@ -172,13 +172,14 @@ import { RoundStatus, TimeLeft } from '@/api/round'
 import {
   LOAD_USER_INFO,
   LOAD_CART,
+  LOAD_COMMITTED_CART_DISPATCH,
```
Comment (samajammin) [inline vue-app/src/components/Cart.vue]: Let's go with "LOAD_COMMITTED_CART"

File: vue-app/src/components/Cart.vue [hunk 4]
```diff
@@ -409,6 +419,19 @@ export default class Cart extends Vue {
   }
 
   getTotal(): BigNumber {
+    // In tally round, use the committedCart to get total contribution amount
+    if (this.$store.state.currentRound.status === RoundStatus.Tallying) {
```
Comment (samajammin) [inline vue-app/src/components/Cart.vue]: Same here - update to any state after reallocation.

**Review by samajammin — CHANGES_REQUESTED** [review]
Looks great for the most part! A few small comments. Reviewed on mobile in between flights so didn't get to all of it.

File: vue-app/src/components/Cart.vue (hunk 1 above)
Comment (corwintines) [reply vue-app/src/components/Cart.vue]: Yeah good call.

**Comment by corwintines** [comment]
@samajammin @pettinarip Took your change requests and put them in. Did a refactor of a mutation-type I put in that wouldnt be needed in the end after looking at Vue.set

**Comment by ryancreatescopy** [comment]
This works as intended for me! Nice work @corwintines

File: contracts/scripts/deployTestRound.ts [hunk 5]
```diff
@@ -59,8 +59,8 @@ async function main() {
   )
   const maciParameters = await MaciParameters.read(maciFactory)
   maciParameters.update({
-    signUpDuration: 300,  // 5 minutes
-    votingDuration: 300,
+    signUpDuration: 75,  // 5 minutes
```
Comment (samajammin) [inline contracts/scripts/deployTestRound.ts]:
```suggestion
    signUpDuration: 300,  // 5 minutes
```

File: contracts/scripts/deployTestRound.ts [hunk 6]
```diff
@@ -59,8 +59,8 @@ async function main() {
...
+    votingDuration: 75,
```
Comment (samajammin) [inline contracts/scripts/deployTestRound.ts]:
```suggestion
    votingDuration: 300,
```

File: vue-app/src/components/ContributionModal.vue [hunk 7]
```diff
@@ -215,6 +215,7 @@ export default class ContributionModal extends Vue {
       return
     }
     this.step += 1
+    this.$store.dispatch(SAVE_COMMITTED_CART_DISPATCH)
```
Comment (samajammin) [inline vue-app/src/components/ContributionModal.vue]: I believe this should only happen when on successful transactions

File: vue-app/src/components/ReallocationModal.vue [hunk 8]
```diff
@@ -80,6 +81,7 @@ export default class ReallocationModal extends Vue {
       return
     }
     this.step += 1
+    this.$store.dispatch(SAVE_COMMITTED_CART_DISPATCH)
```
Comment (samajammin) [inline vue-app/src/components/ReallocationModal.vue]: Same here - I believe this should only happen when on successful transactions

**Review by samajammin — CHANGES_REQUESTED** [review]
Couple small changes.

Overall this looks good! I unfortunately haven't been able to go through the flow locally - I keep getting invalid hex string errors when attempting to contribute (https://github.com/ethereum/clrfund/issues/99).

Let's walk through this flow together on your machine tomorrow so we can get this merged.

**Comment by samajammin** [comment]
FYI @corwintines - I pushed up a small change so make sure you pull before making any commits.

**Review by samajammin — APPROVED** [review]
On 2nd thought, since @ryancreatescopy was able to successfully test interaction, I'm just going to make those small tweaks & merge so we can unblock @pettinarip.

Nice work @corwintines!

**Pull request merged**